lib/challenge.js

```diff
@@ -1,5 +1,31 @@
 'use strict';
 
+//leading 0's are removed
+let oddsOnly = function () {
+  let newArray = [];
+  for (var i = 0; i < 1,000,000; i++) {
+    if (array[i] % 2 === 0) {
+    newArray.pop();
+    } else {
+    newArray.push();
```

Comment:
I'm not really sure what you're doing here with `.push` and `.pop`. Can you explain your approach?